python/cm/models.py

```diff
@@ -201,9 +201,9 @@ class Component(models.Model):
     name = models.CharField(max_length=160)
     display_name = models.CharField(max_length=160, blank=True)
     description = models.TextField(blank=True)
-    params = models.TextField(blank=True)       # JSON
-    constraint = models.TextField(blank=True)   # JSON
-    requires = models.TextField(blank=True)     # JSON
+    params = JSONField(blank=True)
```

Comment:
I don't like this blank=True. This forces us to do a lot of None check in functions. I think, that we should change it to blank=False and set reasonable defaults ([] for example). See my comments above about requires fields.